### PR TITLE
implement batch delete

### PR DESCRIPTION
**Pull request by whyrusleeping**
@raulk im gonna need a solid look-over on this from you as the error handling logic is a bit weird and im not sure how well im handling things.

### CODE REVIEW

File: blockstore.go [hunk 1]
```diff
@@ -504,6 +504,42 @@ Retry:
 	return err
 }
 
+func (b *Blockstore) DeleteMany(cids []cid.Cid) error {
```
Comment (raulk) [inline blockstore.go:507]: Could probably make it a variadic argument.

File: blockstore.go [hunk 2]
```diff
@@ -504,6 +504,42 @@ Retry:
...
+		var err error
+		o.Do(func() { err = b.grow() })
+		if err != nil {
+			return fmt.Errorf("lmdb delete failed: %w", err)
+		}
+		b.oplock.RLock() // reclaim the concurrent lock.
+		goto Retry
+	case lmdb.IsErrno(err, lmdb.ReadersFull):
+		b.oplock.RUnlock() // yield.
+		b.sleep("delete")
```
Comment (raulk) [inline blockstore.go]:
```suggestion
		b.sleep("delete many")
```

File: blockstore.go [hunk 3]
```diff
@@ -504,6 +504,42 @@ Retry:
...
+	b.oplock.RLock()
+	defer b.oplock.RUnlock()
+
+Retry:
+	err := b.env.Update(func(txn *lmdb.Txn) error {
+		for _, cid := range cids {
+			if err := txn.Del(b.db, cid.Hash(), nil); err != nil {
+				return err
+			}
+		}
```
Comment (raulk) [inline blockstore.go]: To not shadow the import:

```suggestion
		for _, c := range cids {
			if err := txn.Del(b.db, c.Hash(), nil); err != nil {
				return err
			}
		}
```

**Review by raulk — COMMENTED** [review]
A test would be welcome 🙏

File: blockstore.go (hunk 1 above)
Comment (whyrusleeping) [reply blockstore.go:507]: Then i wouldnt interfacematch with vyzo: https://github.com/filecoin-project/lotus/blob/master/blockstore/badger/blockstore.go#L355

Comment (raulk) [reply blockstore.go:507]: Ok, right, and go is dumb.

**Review by raulk — APPROVED** [review]

**Pull request merged**